ansible/library/conn_graph_facts.py

```diff
@@ -338,7 +338,19 @@ def main():
             if host_vlan:
                 device_vlan_range.append(host_vlan["VlanRange"])
                 device_vlan_list.append(host_vlan["VlanList"])
-                device_vlan_map_list[hostname] = host_vlan["VlanList"]
+                port_vlans = lab_graph.get_host_port_vlans(hostname)
+                device_vlan_map_list[hostname] = {}
+                for a_host_vlan in host_vlan["VlanList"]:
+                    # Get the corresponding port for this vlan from the port vlan list for this hostname
+                    found_port_for_vlan = False
+                    for a_port in port_vlans:
+                        if a_host_vlan in port_vlans[a_port]['vlanlist']:
+                            port_index = a_port.split("Ethernet")[1]
```

Comment:
`port_index` is actually the interface suffix number, right? Should we use the actual indexes(like 0, 1, 2 ...)?

Reply:
I am not following your question.

I am trying to get the interface suffix using split based on 'Ethernet'.  For the example that I gave above in the PR description,  we would have the following:

```
port_vlans = {
      "Ethernet10": {
         "mode": "Access",
         "vlanids": "120",
         "vlanlist": [
            120
         ]
      },
      "Ethernet11": {
         "mode": "Access",
         "vlanids": "121",
         "vlanlist": [
            121
         ]
      }
   }

host_vlan["VlanList"] = [120, 121]
```

What gets put into the topology file (host_interfaces or vlans under VMs) for the ports Ethernet10 and Ethernet11 are '10' and '11' respectively.

So, we want to match port suffix '10' to vlan '120', and port suffix '11' to vlan '121'.

Reply:
Yes, what if the interfaces are like `Ethernet0`, `Ethernet2`, `Ethernet4`(the suffix numbers are not consecutive).

Reply:
In this case, wouldn't we have '0', '2' and '4' in the topology file, instead of 0, 1, 2?

And the returned device_vlan_map_list would be:
```
"dut1": {
   "0" : 120,
   "2", 121
}
```

Reply:
You are correct - this approach would not work for DUT's that don't have consecutive ports. 

One option that I can see to solve this that we add something similar to port_alias_to_name_map in minigraph_facts to conn_graph_facts which gives the lists of ports based on hwsku.

Would it be ok to do it this way - if so I can code it up.

Reply:
Yes, it will be much more better as indexes are consecutive. With this change, I believe there will be errors in `add-topo`, will you fix those plays?

Reply:
I have committed the code - can you please review again.

Reply:
Making `conn_graph_facts` the ability to get port index from alias seems to introduce code duplication in this way. How about simply make `device_vlan_map_list` simply be like
```
    "dut1" : {
       "Ethernet40" : 120,
       "Ethernet44" : 121
    }
```

Reply:
If we do this, then the problem shifts to generation of 'dut_fp_ports' - specifically in roles/vm_set/library/vlan_port.py. dut_fp_ports eventually is passed into vm_topology where we need indices.

I have another PR to introduce common module_utils - https://github.com/Azure/sonic-mgmt/pull/2623/.  The idea was to put this common code between conn_graph_facts and minigraph_facts into the module_utils.  Once that PR is approved, then would move this common code into module_utils and then push this PR.